applicationset/generators/git.go

```diff
@@ -238,6 +258,40 @@ func (g *GitGenerator) generateParamsFromGitFile(filePath string, fileContent []
 	return res, nil
 }
 
+// filterAppsFiles filters a list of file paths based on inclusion and exclusion patterns defined in GitFileGeneratorItems.
+// It uses doublestar globbing to support complex patterns like "**/*.yaml".
+func (g *GitGenerator) filterAppsFiles(files []argoprojiov1alpha1.GitFileGeneratorItem, allFilePaths []string) []string {
+	res := []string{}
+	for _, filePath := range allFilePaths {
+		appInclude := false
+		appExclude := false
+		for _, file := range files {
+			// since path.Match() doesn't natively support the ** double-star pattern for globbing
+			// we use the doublestar package - https://github.com/bmatcuk/doublestar
+			match, err := doublestar.Match(file.Path, filePath)
+			if err != nil {
+				log.WithError(err).WithField("itemPath", file).
+					WithField("path", filePath).Error("error while matching path to item path")
+				continue
+			}
+			if match && !file.Exclude {
```

Comment:
Do we need two variables? Can't we just do something like 

```
if match && !file.Exclude {
    include = true
}
```

Reply:
Maybe. I will have to think about it. But looking at the code you just wrote, we're required to give priority to the `exclude` that is getting done [here](https://github.com/argoproj/argo-cd/pull/22734/files#diff-e18bda08876f1471ded096f197d58778d250cd942ca66539391b067d5f069f79R285). This is one of the usecases of having two variables.

Reply:
I don't think we need a map either. Please see how I've refactored it and let me know your thoughts.